gcamwrapper/main.py

```diff
@@ -8,41 +9,116 @@ class Gcam (gcam_module.gcam):
        instance.
     """
 
-    def get_data(self, query):
+    def run_to_period(self, period = None):
+        """ Run GCAM up to and including some model period.
+            Model periods which have already been run will be kept track
+            of and will not be run again.  HOWEVER, we do not attempt to
+            keep track of if those model periods are "dirty" such as if
+            a user has called `set_data` in such a way that would invalidate
+            that solution.
+
+        :param period: The model period to run or the `get_current_period` + 1
+                       if None
+        :type period: int
+        """
+
+        if period is None:
+            period = self.get_current_period() + 1
+        super(Gcam, self).run_to_period(period)
+
+    def get_data(self, query, query_params = None):
         """Queries for arbitrary data from a running instance of GCAM.
 
         :param query:   GCAM fusion query
         :type query:    str
+        :param query_params: User options to translate placeholder expressions in query should
+                             it have any
+        :type query_params:  dict(string: array(string))
 
         :returns:       DataFrame with the query results.
 
         """
 
+        units = query.units if hasattr(query, "units") else None
+        if query_params is not None:
+            query = apply_query_params(query, query_params, True)
         data_dict = super(Gcam, self).get_data(query)
         data_df = DataFrame(data_dict)
         # The data comming out of gcam is unaggregated so we will need to do that now
         # first figure out what the "value" column is, group by everything else, and summarize
         # TODO: decide on failure mode when no results
```

Comment:
are you going to raise an exception for this condition?

Reply:
Yea, I'm not sure.  I could raise and exception or just return a data frame with no rows.  The later is what we do in `rgcam` / `gcamreader` so maybe it would be more consistent to do that instead?  I could be convinced either way though.

Reply:
I prefer an informative exception or warning that states something like, "Query did not return any results"